translator/translator.py

```diff
@@ -11,101 +11,109 @@
 debug_mode = os.environ.get("DEBUG")
 if debug_mode:
 
     def install_deps():
-        """
-        Install necessary dependencies for debuggers.
+        """Install necessary dependencies for debuggers.
 
         Because of how we build translator currently, we don't have a great way to selectively
         install things at build, so we just do it here! Right now this also includes base.txt,
         which is unecessary, but in the future when we build a little better, it'll already be
         setup.
         """
-        logging.info("Installing dependencies for debuggers")
+        logger.info("Installing dependencies for debuggers")
 
-        import subprocess
-        import sys
+        import subprocess  # noqa: S404, PLC0415
+        import sys  # noqa: PLC0415
 
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "-r", "/requirements/local.txt"])
+        subprocess.check_call([sys.executable, "-m", "pip", "install", "-r", "/requirements/local.txt"])  # noqa: S603 TODO: add this to the container build
 
-        logging.info("Done installing dependencies for debuggers")
+        logger.info("Done installing dependencies for debuggers")
 
-    logging.info(f"Translator is set to use a debugger. Provided debug mode: {debug_mode}")
+    logger.info("Translator is set to use a debugger. Provided debug mode: %s", debug_mode)
     # We have to setup the debugger appropriately for various IDEs. It'd be nice if they all used the same thing but
     # sadly, we live in a fallen world.
     if debug_mode == "pycharm-pydevd":
-        logging.info("Entering debug mode for pycharm, make sure the debug server is running in PyCharm!")
+        logger.info("Entering debug mode for pycharm, make sure the debug server is running in PyCharm!")
 
         install_deps()
 
         import pydevd_pycharm
 
         pydevd_pycharm.settrace("host.docker.internal", port=56782, stdoutToServer=True, stderrToServer=True)
 
-        logging.info("Debugger started.")
+        logger.info("Debugger started.")
     elif debug_mode == "debugpy":
-        logging.info("Entering debug mode for debugpy (VSCode)")
+        logger.info("Entering debug mode for debugpy (VSCode)")
 
         install_deps()
 
         import debugpy
 
-        debugpy.listen(("0.0.0.0", 56781))
+        debugpy.listen(("0.0.0.0", 56781))  # noqa S104 (doesn't like binding to all interfaces)
 
-        logging.info("Debugger listening on port 56781.")
+        logger.info("Debugger listening on port 56781.")
     else:
-        logging.warning(f"Invalid debug mode given: {debug_mode}. Debugger not started")
+        logger.warning("Invalid debug mode given: %s. Debugger not started", debug_mode)
 
 # Must match the URL in asgi.py, and needs a trailing slash
 hostname = os.environ.get("SCRAM_HOSTNAME", "scram_hostname_not_set")
 url = os.environ.get("SCRAM_EVENTS_URL", "ws://django:8000/ws/route_manager/translator_block/")
 
 
+async def process(message, websocket, g):
```

Comment:
I like this but also does this fit in this MR?

Reply:
This was necessary to reduce the complexity of the function. It was failing because `main` was too complex.